lib/wallaroo/core/source/tcp_source/framed_source_notify.pony

```diff
@@ -134,8 +131,8 @@ class TCPSourceNotify[In: Any val]
         end
 
       if is_finished then
-        let end_ts = Time.nanos()
-        let time_spent = end_ts - ingest_ts
+        let end_ts = WallClock.nanoseconds()
+        let time_spent = end_ts - ingest_ts // I'm unsure of this stuff
```

Comment:
I think you should remove this comment and open an issue to reevaluate the way we do metrics based on what you learned from working on the monitoring hub.  A lot of the metrics code is pretty old and needs reexamination.